cpp/tests/utilities/column_utilities.cu

```diff
@@ -854,6 +854,16 @@ bool expect_columns_equivalent(cudf::column_view const& lhs,
                                fp_ulps);
 }
 
+/**
+ * @copydoc cudf::test::expect_column_empty
+ */
+void expect_column_empty(cudf::column_view const& col)
+{
+  EXPECT_EQ(0, col.size());
+  EXPECT_EQ(0, col.null_count());
+  EXPECT_EQ(0, col.num_children());
```

Comment:
This is surprising to me. Shouldn't a struct column with empty children columns still be considered "empty"?

Reply:
I think  you are right. Looking at `cudf::empty_like` it appears empty compound columns can have children.